service/collector/error.go

```diff
@@ -1,12 +1,47 @@
 package collector
 
 import (
+	"strings"
+
 	"github.com/giantswarm/microerror"
 )
 
+const (
+	// trustedAdvisorUnsupportedErrorMessage is the error message returned
+	// if Trusted Advisor is not supported (support plan is not Business or Enterprise).
+	trustedAdvisorUnsupportedErrorMessage = "AWS Premium Support Subscription is required to use this service."
+)
+
 var invalidConfigError = microerror.New("invalid config")
 
 // IsInvalidConfig asserts invalidConfigError.
 func IsInvalidConfig(err error) bool {
 	return microerror.Cause(err) == invalidConfigError
 }
+
+var invalidResourceError = microerror.New("invalid resource")
+
+// IsInvalidResource asserts invalidResourceError.
+func IsInvalidResource(err error) bool {
+	return microerror.Cause(err) == invalidResourceError
+}
+
+var nilLimitError = microerror.New("nil limit")
+
+// IsNilLimit asserts nilLimitError.
+func IsNilLimit(err error) bool {
+	return microerror.Cause(err) == nilLimitError
+}
+
+var nilUsageError = microerror.New("nil usage")
+
+// IsNilUsage asserts nilUsageError.
+func IsNilUsage(err error) bool {
+	return microerror.Cause(err) == nilUsageError
+}
+
+// IsUnsupportedPlan asserts that an error is due to Trusted Advisor
+// not being available with the current support plan.
+func IsUnsupportedPlan(err error) bool {
+	return strings.Contains(err.Error(), trustedAdvisorUnsupportedErrorMessage)
+}
```

Comment:
Please have a look at https://github.com/giantswarm/fmt/blob/master/go/errors.md. You also find examples here: https://github.com/giantswarm/aws-operator/blob/4c3b165519a0970c908f82bece1c3c484eb30a1d/service/controller/v14/resource/s3bucket/error.go#L25-L44, or here: https://github.com/giantswarm/errors/blob/55d763df908376a858290cf4009e8bd1c768ccef/guest/error.go#L52-L78.